From fe688388f4dbc2891eba199d8f793739cad324f5 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Fri, 8 Feb 2019 14:55:48 +0000
Subject: [PATCH 2/2] Increase the amount of time for the inotify event to
 happen

On slow hardware 2 seconds (!) still isn't enough.

Maybe fixes the other half of https://github.com/hughsie/libxmlb/issues/10
---
 src/xb-self-test.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/xb-self-test.c b/src/xb-self-test.c
index cc131eb..bffcf0e 100644
--- a/src/xb-self-test.c
+++ b/src/xb-self-test.c
@@ -22,6 +22,8 @@
 static GMainLoop *_test_loop = NULL;
 static guint _test_loop_timeout_id = 0;
 
+#define XB_SELF_TEST_INOTIFY_TIMEOUT		10000 /* ms */
+
 static gboolean
 xb_test_hang_check_cb (gpointer user_data)
 {
@@ -478,7 +480,7 @@ xb_builder_ensure_watch_source_func (void)
 	ret = g_file_set_contents ("/tmp/temp.xml", "<id>inkscape</id>", -1, &error);
 	g_assert_no_error (error);
 	g_assert_true (ret);
-	xb_test_loop_run_with_timeout (2000);
+	xb_test_loop_run_with_timeout (XB_SELF_TEST_INOTIFY_TIMEOUT);
 	g_assert_false (xb_silo_is_valid (silo));
 	g_assert_cmpint (invalidate_cnt, ==, 1);
 	g_assert_false (xb_silo_is_valid (silo));
@@ -542,7 +544,7 @@ xb_builder_ensure_func (void)
 				       G_FILE_CREATE_NONE, NULL, NULL, &error);
 	g_assert_no_error (error);
 	g_assert_true (ret);
-	xb_test_loop_run_with_timeout (2000);
+	xb_test_loop_run_with_timeout (XB_SELF_TEST_INOTIFY_TIMEOUT);
 	g_assert_false (xb_silo_is_valid (silo));
 	g_assert_cmpint (invalidate_cnt, ==, 1);
 
-- 
2.7.4

